Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.java

```diff
@@ -297,6 +299,15 @@ public boolean onTouch(View v, MotionEvent event)
           }
           break;
       }
+
+      // Not all devices that support VibrationEffect (API>=26) allow Amplitude Control
+      if (mPreferences.getInt("hapticFeedbackLevel", 0) >= 1 &&
+        android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
+        Vibrator vibratorManager = (Vibrator)
+          getContext().getSystemService(Context.VIBRATOR_SERVICE);
+        vibratorManager.vibrate(VibrationEffect.createOneShot(80,
+          mPreferences.getInt("hapticFeedbackLevel", 1)));
+    }
```

Comment:
You should include a fallback for older API levels. Preferably vibrating without amplitude control, or if that isn't appropriate, making the option not show up in the GUI.